src/app/account/account-report.service.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { AccountReport } from '../shared/models/account-report.model';
+import { environment } from '../../env/environment';
+import { ApiPaths } from '../shared/api/api-paths.enum';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AccountReportService {
+  constructor(private http: HttpClient) { }
+
+  getReviewReportById(id: number): Observable<AccountReport> {
+    return this.http.get<AccountReport>(`${environment.apiHost}/${ApiPaths.ReportedAccounts}/${id}`);
+  }
+
+  getAllAccountReports(): Observable<AccountReport[]> {
+    return this.http.get<AccountReport[]>(`${environment.apiHost}/${ApiPaths.ReportedAccounts}`);
+  }
+  blockAccount(id: number): Observable<void> {
```

Comment:
I recommend changing the name of the parameter to reportId because the `id` in question is the id of the report for that account.
```suggestion
  blockAccount(reportId: number): Observable<void> {
```